### PR TITLE
🥗🧹 `Marketplace`: Extract and test `MarketplaceComponent`

### PR DESCRIPTION
**Pull request by zspencer**
- https://github.com/zinc-collective/convene/issues/1187
- https://github.com/zinc-collective/convene/issues/1326

I'm not particularly happy with how I'm doing the render test... But calling `output.blank?` didn't work; because apparently a fragment counts as not-blank, even if it's empty...

### CODE REVIEW

File: app/furniture/marketplace/marketplace_component.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+class Marketplace
+  class MarketplaceComponent < ApplicationComponent
+    attr_accessor :marketplace
+    delegate :location, :carts, :ready_for_shopping?, to: :marketplace
+
+    def initialize(marketplace:, data: {}, classes: "", **kwargs)
+      super(data: data, classes: classes, **kwargs)
```
Comment (anaulin) [inline app/furniture/marketplace/marketplace_component.rb:7]: I think you could just leave the `data` and `classes` parameters to be part of `**kwargs` and not have to deal with them explicitly.

Relatedly, recently I came across `dry-initializer`, which the Evil Martians folks seem to like to use for their ViewComponents, and I am wondering if that would be a good way for us to tighten up our component constructors as well:
https://dry-rb.org/gems/dry-initializer/3.0/inheritance/
https://github.com/palkan/view_component-contrib#hanging-initialize-out-to-dry

File: app/furniture/marketplace/marketplace_component.html.erb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<% if !ready_for_shopping? && policy(marketplace).edit? %>
+  <%= render AlertComponent.new(scheme: :warning, icon: :exclamation_triangle,
+                                title: t('marketplace.marketplace.config_missing_title')) do %>
+    <%= t('marketplace.marketplace.config_missing_explainer_html',
+          edit_link: link_to(t('marketplace.marketplace.edit'), location(:edit)) ) %>
+  <% end %>
+<% end %>
+
+<%- if ready_for_shopping? || policy(marketplace).edit? %>
```
Comment (anaulin) [inline app/furniture/marketplace/marketplace_component.html.erb:9]: This is now the same as `render?`, so we could remove the `if` altogether.

**Review by anaulin — APPROVED** [review]
Beautiful, thanks!

File: app/furniture/marketplace/marketplace_component.rb (hunk 1 above)
Comment (zspencer) [reply app/furniture/marketplace/marketplace_component.rb:7]: Ooo, I do like a lot of the the Evil Martians gems! I'd be down to give it a shot!

**Pull request merged**